test/xpu_api/random/unit_tests/philox_unit_test.pass.cpp

```diff
@@ -338,42 +362,99 @@ template <typename Engine>
 int
 discard_overflow_test()
 {
+    int ret_sts = 0;
+
     using T = typename Engine::result_type;
     using scalar_type = typename Engine::scalar_type;
 
-    Engine engine1;
-    std::array<T, Engine::word_count> counter;
-
-    for (int i = 0; i < Engine::word_count; i++)
+    // Iterate through the counter's position being overflown
+    for (int overflown_position = 0; overflown_position < Engine::word_count - 1; ++overflown_position)
     {
-        counter[i] = 0;
-    }
+        Engine engine1;
+        std::array<T, Engine::word_count> counter = {0};
 
-    // overflow of the 0th counter element
-    counter[2] = 1;
+        // Overflow of a counter's element. The correspondence is the following:
+        // 0 1 2 3 overflow position
+        // 1 2 3 0 set-up counter position in engine
+        // 2 1 0 3 raw_counter_position
+        int raw_counter_position = (Engine::word_count - overflown_position - 2) % Engine::word_count;
```

Comment:
please, check this line if `Engine::word_count - overflown_position - 2` would overflow or is it expected?

also is -2 here only for Engine::word_count = 4?

Reply:
> please, check this line if Engine::word_count - overflown_position - 2 would overflow or is it expected?

This question was discussed offline.

> also is -2 here only for Engine::word_count = 4?

I've double-checked, this equation is also valid for `word_count = 2`